sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -88,6 +88,33 @@ internal static void WrapAsyncResultCore<T>(IInvocation invocation, Type generic
                 async ValueTask<T> Await()
                 {
                     invocation.Proceed();
+                    var tType = typeof(T);
+                    if (tType.IsGenericType && tType.GetGenericTypeDefinition().Equals(typeof(Response<>)))
+                    {
+                        //guaranteed only one generic arg with Response<T>
+                        var tGenericType = tType.GetGenericArguments()[0];
+                        var returnType = invocation.ReturnValue.GetType();
```

Comment:
Can ReturnValue be null here?

Reply:
99.9% sure it can't but I can add the extra check.  If it is the original cast wouldn't work